src/Simplify.cpp

```diff
@@ -4712,6 +4777,76 @@ void check_bounds() {
     check(min(2, x) - x, 2 - max(x, 2));
     check(max(2, x) - x, 2 - min(x, 2));
 
+    check(max(min(max(x, y), z), y), min(max(x, y), z));
+    check(max(min(max(y, x), z), y), min(max(y, x), z));
+    check(min(max(min(x, y), z), y), max(min(x, y), z));
+    check(min(max(min(y, x), z), y), max(min(y, x), z));
+
+    {
+        Expr one = broadcast(cast(Int(16), 1), 64);
+        Expr three = broadcast(cast(Int(16), 3), 64);
+        Expr four = broadcast(cast(Int(16), 4), 64);
+        Expr five = broadcast(cast(Int(16), 5), 64);
+        Expr neg_four = broadcast(cast(Int(16), -4), 64);
+        Expr neg_five = broadcast(cast(Int(16), -5), 64);
+        Expr v1 = Variable::make(Int(16).with_lanes(64), "x");
+        Expr v2 = Variable::make(Int(16).with_lanes(64), "y");
+
+        // min(max(min(x, 4), -4), 4) -> max(min(x, 4), -4)
+        check(min(max(min(v1, four), neg_four), four), max(min(v1, four), neg_four));
```

Comment:
Maybe we should check all 4 possible orderings for a max/min pair and their arguments? I'm wondering about const_int_bounds missing two of the 4 orderings, I wanted to at least make sure that it hits the ones generated by nested clamps.

Or maybe check clamp in addition to these tests, instead of replacing them.